### DISCUSSION TITLE
alarm_control_panel support "instant" arming types

### DISCUSSION THREAD

**Discussion by austinmroczek** [home-assistant/architecture#656] [Entity Models]
Many alarms provide an "instant" arming method.  This means the alarm will instantly trigger and call the police if a door is opened.  This is different than "normal" arming where there is a 30 or 60 second delay between the door opening and the alarm triggering, giving a user time to disarm the system by putting in their code.  Many people use the instant arm away when they are on vacation to reduce police response time, or use the instant arm home at night when the entire family is already home.

I propose adding to the `AlarmControlPanelEntity` functions for `alarm_arm_away_instant` and `alarm_arm_home_instant`, adding services of the same names (and all the associated "async_" versions).  It would also require new `SUPPORT_ALARM_ARM_AWAY_INSTANT` and similar constants.

For the front end, I assume this would mean either adding additional buttons for the arm instant functions; adding a checkbox for "instant" next to Arm Home; or some other way to show the user this option. 

This is a feature specifically requested for TotalConnect in https://github.com/home-assistant/core/issues/59435 but I'm pretty sure many alarm systems have a similar concept.

I can create a PR for Core but would need help for the Front end piece.

**Comment by MartinHjelmare** [comment]
Please do some research and list how many integrations would benefit from the new services. We only add new features to entity integrations if many platforms can use them.

**Reply by austinmroczek** [reply]
I looked up ~50% of the 36 alarm integrations.  I found 3 that the concept definitely applies to.  There are 7 more that have selectable entry delay for the physical panel, but it is unclear if that controllable by the APIs.  

Given how many Honeywell/Ademco/etc panels are supported by TotalConnect and have this concept I figured this would be more widespread.  I'll implement this for `totalconnect` and others can copy it if desired.

**Reply by tdejneka** [reply]
The [Elk M1 integration](https://www.home-assistant.io/integrations/elkm1/#services) provides custom services to enable instant night and instant home modes.

- elkm1.alarm_arm_home_instant
- elkm1.alarm_arm_night_instant

Obviously, the integration currently has to map those modes, to one of the supported modes, but it would benefit from the addition of instant mode(s).

**Reply by austinmroczek** [reply]
I am almost done implementing the concept for TotalConnect at https://github.com/home-assistant/core/pull/60156 if anyone wants to see it.

Adding the service calls was easy.  The only question will be if/how to add buttons for the UI.
